psh/date/date.c

```diff
@@ -0,0 +1,223 @@
+/*
+ * Phoenix-RTOS
+ *
+ * date - print or set the system date and time
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Mateusz Niewiadomski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <sys/time.h>
+#include <unistd.h>
+
+#include "../psh.h"
+
+
+enum mode { mode_skip,
+	mode_help,
+	mode_print,
+	mode_set,
+	mode_parse };
+
+
+static void psh_date_info(void)
+{
+	printf("print/set the system date and time");
+}
+
+
+static void psh_date_help(const char *prog)
+{
+	printf(
+		"Usage: %s [-h] [-s EPOCH] [-d @EPOCH] [+FORMAT]\n"
+		"  -h:  shows this help message\n"
+		"  -s:  set system time described by EPOCH (POSIX time format)\n"
+		"  -d:  display time described by EPOCH (POSIX time format)\n"
+		"  FORMAT: string with POSIX date formatting characters\n"
+		"NOTE: FORMAT string not supported by options: '-s', '-d'\n",
+		prog);
+}
+
+
+static int psh_date_print(const struct timeval *tv, const char *format)
+{
+	struct tm *tinfo;
+	const char *defformat = "+%a, %d %b %y %H:%M:%S";
+	char buf[64];
+
+	if (format == NULL) {
+		format = defformat;
+	}
+	else if (format[0] != '+') {
+		fprintf(stderr, "date: invalid format '%s'\n", format);
+		return 1;
+	}
+
+	tinfo = localtime(&(tv->tv_sec));
```

Comment:
this may return NULL on error, I don't know if it's relevant

Reply:
Just checked documentation - `localtime()` should be able to return `NULL` under some circumstances. I did not knew that. I only checked just our implementation which (apparently) cant return `NULL` under any circumstances. It is pure mathematics based on `const time_t *timer` argument. 

Added `NULL` check just to be future-safe. Thanks!